### PR TITLE
Migrate API usage from v1 to v2

### PR DESCRIPTION
**Pull request by biggless**

### CODE REVIEW

File: lib/lita/commands/notes.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
+module Commands
+  class Notes
+    include Base
+
+    def call
+      incident_id = message.match_data['incident_id']
+      notes = pagerduty.get_notes_by_incident_id(incident_id)
+      response format_notes(notes, incident_id)
+    rescue Exceptions::IncidentNotFound
+      response message: 'incident.not_found', params: { id: incident_id }
```
Comment (EugeneOsadchyi) [inline lib/lita/commands/notes.rb]:
```suggestion
      response message: 'incident.not_found',
               params: { id: incident_id }
```

File: lib/lita/commands/incident.rb [hunk 2]
```diff
@@ -0,0 +1,13 @@
+module Commands
+  class Incident
+    include Base
+
+    def call
+      incident_id = message.match_data['incident_id']
+      incident = pagerduty.get_incident(incident_id)
+      response format_incidents([incident])
+    rescue Exceptions::IncidentNotFound
+      response message: 'incident.not_found', params: { id: incident_id }
```
Comment (EugeneOsadchyi) [inline lib/lita/commands/incident.rb]:
```suggestion
      response message: 'incident.not_found',
               params: { id: incident_id }
```

File: lib/lita/commands/ack_all.rb [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+    rescue Exceptions::IncidentsEmptyList
+      response message: 'incident.none'
+    rescue Exceptions::IncidentManageUnsuccess
+      nil
+    end
+
+    private
+
+    def query_params
+      { 'statuses' => %w[triggered acknowledged] }
```
[the remark this answers is not in the document]
Comment (EugeneOsadchyi) [reply lib/lita/commands/ack_all.rb]:
```suggestion
      { statuses: %w[triggered acknowledged] }
```

File: lib/lita/commands/ack_mine.rb [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+      response message: 'incident.none_mine'
+    rescue Exceptions::IncidentManageUnsuccess
+      nil
+    end
+
+    private
+
+    def query_params
+      {
+        'statuses' => %w[triggered acknowledged],
```
Comment (EugeneOsadchyi) [inline lib/lita/commands/ack_mine.rb]:
```suggestion
        statuses: %w[triggered acknowledged],
```

File: lib/lita/commands/on_call_lookup.rb [hunk 5]
```diff
@@ -0,0 +1,41 @@
+module Commands
+  class OnCallLookup
+    include Base
+
+    def call
+      response message: 'on_call_lookup.response', params: success_params
```
Comment (EugeneOsadchyi) [inline lib/lita/commands/on_call_lookup.rb]:
```suggestion
      response message: 'on_call_lookup.response',
               params: success_params
```

File: lib/lita/commands/resolve_all.rb [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+    rescue Exceptions::IncidentsEmptyList
+      response message: 'incident.none'
+    rescue Exceptions::IncidentManageUnsuccess
+      nil
+    end
+
+    private
+
+    def query_params
+      { 'statuses' => %w[triggered acknowledged] }
```
Comment (EugeneOsadchyi) [inline lib/lita/commands/resolve_all.rb]:
```suggestion
      { statuses: %w[triggered acknowledged] }
```

File: lib/lita/commands/resolve_mine.rb [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+      response message: 'incident.none_mine'
+    rescue Exceptions::IncidentManageUnsuccess
+      nil
+    end
+
+    private
+
+    def query_params
+      {
+        'statuses' => %w[triggered acknowledged],
```
Comment (EugeneOsadchyi) [inline lib/lita/commands/resolve_mine.rb]:
```suggestion
        statuses: %w[triggered acknowledged],
```

File: spec/pagerduty_spec.rb [hunk 8]
```diff
@@ -0,0 +1,106 @@
...
+    expect{ pagerduty.get_notes_by_incident_id('ABC123') }.to raise_exception Exceptions::NotesEmptyList
+  end
+
+  it 'manage_incidents' do
+    params = { incidents: [
+      { id: 'a', status: 'acknowledged', type: 'incident_reference' },
+      { id: 'b', status: 'acknowledged', type: 'incident_reference' }
+    ]}
+    expect(http).to receive(:put).with('/incidents', params).and_return(
+      OpenStruct.new(body: { users: [] }.to_json, status: 200)
```
Comment (EugeneOsadchyi) [inline spec/pagerduty_spec.rb]:
```suggestion
      OpenStruct.new(body: { users: []}.to_json, status: 200)
```

File: lib/lita/commands/ack_all.rb (hunk 3 above)
[the remark this answers is not in the document]
Comment (biggless) [reply lib/lita/commands/ack_all.rb]: applied

File: lib/lita/commands/notes.rb (hunk 1 above)
Comment (biggless) [reply lib/lita/commands/notes.rb]: Any profit of this?

Comment (EugeneOsadchyi) [reply lib/lita/commands/notes.rb]: No profit.
Just saw it your code different styles one/multi-lines for `response`

It does not affect anything

Comment (biggless) [reply lib/lita/commands/notes.rb]: it depends on line length :)

**Comment by tristaneuan** [comment]
Thanks for the contribution! Will merge and release as 1.0.0

**Pull request merged**